gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,398 @@
+          namespace.levels().length == 0 ? "empty" : namespace.toString(),
+          e);
+      return false;
+    }
+
+    HiveDatabaseOptions.Builder optionsBuilder = builder.getHiveOptionsBuilder();
+    properties.forEach(optionsBuilder::putParameters);
+    client.updateDatabaseParameters(getDatabaseName(namespace), optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> properties) {
+    Database.Builder builder;
+    try {
+      builder = getDatabase(namespace).toBuilder();
+    } catch (IllegalArgumentException e) {
+      LOG.warn(
+          "removeProperties is only supported for tables and databases, namespace {} is not"
+              + " supported",
+          namespace.levels().length == 0 ? "empty" : namespace.toString(),
+          e);
+      return false;
+    }
+
+    HiveDatabaseOptions.Builder optionsBuilder = builder.getHiveOptionsBuilder();
+    properties.forEach(optionsBuilder::removeParameters);
+    client.updateDatabaseParameters(getDatabaseName(namespace), optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    if (namespace.levels().length == 0) {
+      // Calls getCatalog to check existence. BLMS catalog has no metadata today.
+      client.getCatalog(catalogName);
+      return new HashMap<String, String>();
+    } else if (namespace.levels().length == 1) {
+      return getMetadata(getDatabase(namespace));
+    } else {
+      throw new IllegalArgumentException(invalidNamespaceMessage(namespace));
+    }
+  }
+
+  @Override
+  public String name() {
+    return catalogPulginName;
+  }
+
+  @Override
+  protected Map<String, String> properties() {
+    return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
+  }
+
+  @Override
+  public void setConf(Object conf) {
+    this.conf = conf;
+  }
+
+  private String getDatabaseLocation(String dbId) {
+    String warehouseLocation =
+        LocationUtil.stripTrailingSlash(
+            catalogProperties.get(CatalogProperties.WAREHOUSE_LOCATION));
+    Preconditions.checkNotNull(warehouseLocation, "Data warehouse location is not set");
+    return String.format("%s/%s.db", LocationUtil.stripTrailingSlash(warehouseLocation), dbId);
+  }
+
+  private static TableIdentifier getTableIdentifier(Table table) {
+    TableName tableName = TableName.parse(table.getName());
+    return TableIdentifier.of(Namespace.of(tableName.getDatabase()), tableName.getTable());
+  }
+
+  private static Namespace getNamespace(Database db) {
+    return Namespace.of(DatabaseName.parse(db.getName()).getDatabase());
+  }
+
+  private TableName getTableName(String dbId, String tableId) {
+    return TableName.of(projectId, location, catalogId, dbId, tableId);
+  }
+
+  private String getDatabaseId(Namespace namespace) {
+    Preconditions.checkArgument(
```

Comment:
This isn't the right exception. If the namespace is longer than 1, then it should throw `NoSuchNamespaceException`. If the namespace is empty, then this should just return null, I think.

Reply:
Changed it to throw NoSuchNamespaceException.

databaseId() is a util to extract a string database ID. Returning a null here will crash at other places that use its returning value, e.g., `DatabaseName.of(projectId, region, catalogId, databaseId(namespace))`. All valid cases allowing empty namespace has been explicitly checked by `if (namespace.levels().length == 0)`, so I think we are good.